src/chainstate/stacks/boot/pox-2.clar

```diff
@@ -269,34 +271,42 @@
 ;;  the pox-addr was added to the given cycle. 
 (define-private (add-pox-addr-to-ith-reward-cycle (cycle-index uint) (params (tuple 
                                                             (pox-addr (tuple (version (buff 1)) (hashbytes (buff 20))))
+                                                            (reward-set-indexes (list 12 uint))
                                                             (first-reward-cycle uint)
                                                             (num-cycles uint)
+                                                            (stacker (optional principal))
                                                             (amount-ustx uint)
                                                             (i uint))))
     (let ((reward-cycle (+ (get first-reward-cycle params) (get i params)))
           (num-cycles (get num-cycles params))
-          (i (get i params)))
+          (i (get i params))
+          (reward-set-index (if (< i num-cycles)
+            (let ((total-ustx (get-total-ustx-stacked reward-cycle))
+                  (reward-index
+                      ;; record how many uSTX this pox-addr will stack for in the given reward cycle
+                      (append-reward-cycle-pox-addr
+                        (get pox-addr params)
+                        reward-cycle
+                        (get amount-ustx params)
+                        (get stacker params)
+                        )))
+                  ;; update running total
+                  (map-set reward-cycle-total-stacked
+                     { reward-cycle: reward-cycle }
+                     { total-ustx: (+ (get amount-ustx params) total-ustx) })
+                  (some reward-index))
+            none))
+          (next-i (if (< i num-cycles) (+ i u1) (+ i u0))))
```

Comment:
`(+ i u0)`?  Why not just `i`?

Reply:
I can update -- this was just reformatting of the existing pox-1 code